app/controllers/campaign_pages_controller.rb

```diff
@@ -14,13 +14,13 @@ def new
   end
 
   def create
-    @campaign_page = CampaignPage.new(@page_params)
-    @campaign_page.compile_html
-    if @campaign_page.save
-      redirect_to @campaign_page, notice: 'Campaign page updated!'
-    else
+    page_builder = CampaignPageBuilder.new(CampaignPage.new, @page_params, params)
+    if page_builder.switched_template? or not page_builder.save
```

Comment:
@osahyoun what do you think of this? I think I'm going to leave this branch as is, and you guys can make any edits you think would be good before you merge.